pulsar-client-cpp/tests/ClientTest.cc

```diff
@@ -114,3 +116,50 @@ TEST(ClientTest, testConnectTimeout) {
     clientLow.close();
     clientDefault.close();
 }
+
+TEST(ClientTest, testGetNumberOfReferences) {
+    Client client("pulsar://localhost:6650");
+
+    // Producer test
+    uint64_t numberOfProducers = 0;
+    const std::string nonPartitionedTopic = "nonPartitionedTopic";
+    const std::string partitionedTopic = "partitionedTopic";
```

Comment:
Can you modify these two topic name to make sure they're unique and can be recognized as the related topics of `testGetNumberOfReferences`? You can take `ConsumerTest.testIsConnected` as example.